webpack.config.js

```diff
@@ -0,0 +1,147 @@
+const path = require("path")
+const HTMLWebpackPlugin = require("html-webpack-plugin")
+const { CleanWebpackPlugin } = require("clean-webpack-plugin")
+const CopyPlugin = require("copy-webpack-plugin")
+const MiniCssExtractPlugin = require("mini-css-extract-plugin")
+const CssMinimizerPlugin = require("css-minimizer-webpack-plugin")
+const TerserPlugin = require('terser-webpack-plugin')
+
+const isDev = process.env.NODE_ENV === "development"
+const isProd = !isDev
+
+const optimization = () => {
+    const config = {
+        splitChunks: {
+            // chunks: "all",
+            cacheGroups: {
+                commons: {
+                    test: /[\\/]node_modules[\\/]/,
+                    name: 'vendors',
+                    chunks: 'all',
+                },
+            },
+        },
+    }
+
+    if (isProd) {
+        config.minimizer = [
+            new CssMinimizerPlugin(),
+            new TerserPlugin()
+        ]
+    }
+
+    return config
+}
+
+const filename = ext => isDev ? `[name].${ext}` : `[name][fullhash].${ext}`
+
+module.exports = {
+    context: path.resolve(__dirname, "src"),
+    mode: "development",
+    entry: ["@babel/polyfill", "./index.tsx"],
+    output: {
+        filename: filename('js'),
+        path: path.resolve(__dirname, "dist")
+    },
+    devServer: {
+        port: 8000,
+        hot: isDev
+    },
+    devtool: isDev ? "source-map" : "eval",
+    plugins: [
+        new HTMLWebpackPlugin({
+            template: "./index.html",
+            minify: {
+                collapseWhitespace: isProd
+            }
+        }),
+        new CleanWebpackPlugin(),
+        new CopyPlugin({
+            patterns: [
+                { from: "static", to: "static" }
+            ],
+        }),
+        new MiniCssExtractPlugin({
```

Comment:
I don't think it makes much sense to optimize your CSS in development mode, since it will only slow down hot reloads unnecessarily.